### PR TITLE
Storage.basedir configuration improvements

### LINKED ISSUE #1129: Re-work the MutableStorage to not use System.getProperties, but the Spring Boot Environment instead
# Task Description

As a continuation of #1023, we need to re-work the `MutableStorage` class to not use `System.getProperties` to work out the `basedir` of the respective storage.

The folllowing code should be dropped/re-worked:

```
public void initDefaultBasedir(String id)
{
    //TODO: we should rework this to use SpringBoot environment instead of `System.getProperty`
    String storagesBaseDir = System.getProperty("strongbox.storage.booter.basedir");
    Assert.notNull(storagesBaseDir, "System property `strongbox.storage.booter.basedir` should be configured.");
        
    Path basedirPath = Paths.get(storagesBaseDir);
    basedir = basedirPath.resolve(id).toString();
}
```

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

### PR DESCRIPTION
**Pull request by sbespalov**
Fixes #1129 
Contains:
- `TestRepository` and `TestArtifact` annotations improvements
- `MutableStorage.initDefaultBasedir` method dropped
- `Storage.getBasedir()` usage dropped from tests
- partially fix for #1165

### CODE REVIEW

**Comment by sbespalov** [comment]
@carlspring this PR ready
[last 5](https://jenkins.carlspring.org/job/strongbox/job/strongbox-os-builds/) OS builds was green
[s-w-i-t](https://jenkins.carlspring.org/job/strongbox/job/strongbox-web-integration-tests-from-sources/102/) green too

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/RepositoryPath.java [hunk 1]
```diff
@@ -177,6 +179,10 @@ public RepositoryPath resolveSibling(Path other)
 
     protected Path unwrap(Path other)
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/RepositoryPath.java]: I'm sorry... I think I've asked this before, but what was this unwrapping for?

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java [hunk 2]
```diff
@@ -329,18 +329,18 @@ private void setAllows()
                              {
                                  if (storage.getRepositories() != null && !storage.getRepositories().isEmpty())
                                  {
-                                     for (MutableRepository repository : storage.getRepositories().values())
+                                     for (Repository repository : storage.getRepositories().values())
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java]: Wouldn't it be better to do:
```
for (Repository r : storage.getRepositories().values())
{
    MutableRepository repository = (MutableRepository) repository;
    ...
}
```

And use it like this? If I recall correctly, casting to a sub-class was actually expensive.
(I might be wrong, but it also looks a bit unpleasant).

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/RepositoryPath.java (hunk 1 above)
Comment (sbespalov) [reply strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/RepositoryPath.java]: with this method you are able to safe get original `Path` instance from `RepositoryPath`.

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java (hunk 2 above)
Comment (sbespalov) [reply strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java]: fixed

**Comment by carlspring** [comment]
Thanks, @sbespalov! :)

**Pull request merged**